shell/platform/android/vsync_waiter_android.cc

```diff
@@ -66,8 +90,41 @@ void VsyncWaiterAndroid::OnVsyncFromNDK(int64_t frame_nanos, void* data) {
                    "frame_target_time",
                    target_time.ToEpochDelta().ToMicroseconds());
 
-  auto* weak_this = reinterpret_cast<std::weak_ptr<VsyncWaiter>*>(data);
-  ConsumePendingCallback(weak_this, frame_time, target_time);
+  auto* weak_this = reinterpret_cast<std::weak_ptr<VsyncWaiterAndroid>*>(data);
+  ConsumePendingCallback(weak_this, frame_time, target_time, target_time, 0);
+}
+
+// static
+void VsyncWaiterAndroid::OnVsyncFromNDK33(
+    AndroidChoreographer::AChoreographerFrameCallbackData* callback_data,
+    void* data) {
+  auto frame_time = AndroidChoreographer::GetFrameTime(callback_data);
+  auto now = fml::TimePoint::Now();
+  if (frame_time > now) {
```

Comment:
Please add a comment explaining why we are modifying the frame time here.

Reply:
I don't know why this is done, I just copied it from the previous code. And to be honest, it reminds me of some confusion all over the code base, where different timespecs are mixed. So it may be worth looking at all those and cleaning it up.

Reply:
Looked into it more and put a comment to explain why, sometimes Android may give you a timestamp which is a tiny bit in the future, and the rest of the engine can't handle that.